app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/SearchResultsAdapter.java

```diff
@@ -44,54 +44,21 @@ public SearchResultHolder onCreateViewHolder(ViewGroup parent, int viewType) {
 
     @Override
     public void onBindViewHolder(final SearchResultHolder holder, int position) {
-        Datum datum = datumList.get(position);
+        final Datum datum = datumList.get(position);
+        holder.previewImageView.setVisibility(View.GONE);
         if (datum != null) {
             holder.titleTextView.setText(Html.fromHtml(datum.getTitle()));
             holder.descriptionTextView.setText(Html.fromHtml(datum.getDescription()));
-            if (!TextUtils.isEmpty(datum.getLink())) {
-                LinkPreviewCallback linkPreviewCallback = new LinkPreviewCallback() {
-                    @Override
-                    public void onPre() {
-                        holder.previewImageView.setVisibility(View.GONE);
-                        holder.descriptionTextView.setVisibility(View.GONE);
-                        holder.titleTextView.setVisibility(View.GONE);
-                        holder.previewLayout.setVisibility(View.GONE);
+            holder.previewLayout.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    Uri webpage = Uri.parse(datum.getLink());
+                    Intent intent = new Intent(Intent.ACTION_VIEW, webpage);
+                    if (intent.resolveActivity(context.getPackageManager()) != null) {
+                        context.startActivity(intent);
                     }
-
-                    @Override
-                    public void onPos(final SourceContent sourceContent, boolean b) {
-                        holder.previewLayout.setVisibility(View.VISIBLE);
-                        holder.previewImageView.setVisibility(View.VISIBLE);
-                        holder.descriptionTextView.setVisibility(View.VISIBLE);
-                        holder.titleTextView.setVisibility(View.VISIBLE);
-                        holder.titleTextView.setText(sourceContent.getTitle());
```

Comment:
>why we need preview 
Suppose I searched for something which has 3 meanings. Now server responses with these 3 search results and according to you we just display link of these 3 results as provided by server. So, now user has to open the link in browser each time to see what this link is about. Now, what if we show a little preview of that link in the chat itself so that user knows what will be there if he clicks on that link. This approach is similar to google assistant.
>In rss we have to use result from server.
We have to use the result of server and we are using it.
>There is a link in server we can use that link to show result in browser.
We are using this link only to display preview and it will also be used when user clicks on the preview to open it in browser.

See there are 3 things we are getting from server : title, description, link. Display title as it is. Display description as it is. Display link as it is. Also in addition, display preview of that link. This is all which is needed to be done.